crates/runtime/src/databackend/duckdb.rs

```diff
@@ -168,6 +171,7 @@ impl<'a> DuckDBUpdate<'a> {
             self.insert_batch(batch)?;
         }
 
+        self.monitor_duckdb()?;
```

Comment:
I don't think this is the best way to, on a frequent schedule, to get metrics about the specific backend

Reply:
No, this should be on a timer that only gets started if the metrics endpoint is enabled.

Reply:
We may want to extend the trait definition to include a function that collects metrics that the runtime calls in the timer loop for each backend.